### PR TITLE
Fix to race condition in spectator

### PR DESCRIPTION
**Pull request by ChrSacher**

### CODE REVIEW

**Comment by thecogdoctor** [comment]
Please detail changes, additions fixes, etc in the changelog.

File: uo_sys_framework/Core/cfgFunctions/EndConditions/defines/Category1Loop.hpp [hunk 1]
```diff
@@ -225,28 +225,35 @@ if (GETMVAR(Enabled_1,false)) then {
                 };
 
                 if (_ExtractionCheck) then {
-                    if (GVAR(Mode_1) isEqualto 1) then {
-                        {
-                            _x params ["_name","_value"];
-                            if (_value) exitwith {
-                                LOG_1("Category 1 Ending due to :%1",_value);
+                    if((GETMVAR(ExtractionEnabled_1,false)) && (GETMVAR(EndConditions,ExtractionForced_1,false))) then
```
Comment (PiZZAD0X) [inline uo_sys_framework/Core/cfgFunctions/EndConditions/defines/Category1Loop.hpp]: `GETMVAR(EndConditions,ExtractionForced_1,false)` should be `GETMVAR(ExtractionForced_1,false)`

File: uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf [hunk 2]
```diff
@@ -5,7 +5,7 @@ EXEC_CHECK(ALL);
 params ["_unit"];
 
 LOG("Killed_Event called");
-[QGVAR(KilledEvent), _unit] call CBA_fnc_serverEvent;
+//[QGVAR(KilledEvent), _unit] call CBA_fnc_serverEvent;
```
Comment (PiZZAD0X) [inline uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: These events are for use later in wave respawn/player dead handling of counts and unit vars by the server

Comment (ChrSacher) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: So it is intended that the killed functions on the server are called twice?

Comment (PiZZAD0X) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: the XEH fires local to unit

Comment (ChrSacher) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: GVAR(EventRespawnedHandle) = addMissionEventHandler ["EntityRespawned", {_this call FUNC(EventRespawned);}];
GVAR(EventKilledHandle) = addMissionEventHandler ["EntityKilled", {_this call FUNC(EventKilled);}];
GVAR(EventDisconnectHandle) = addMissionEventHandler ["HandleDisconnect", {_this call FUNC(EventDisconnect);}];

Comment (ChrSacher) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: These are also there calling the exact same function. Don't really know.

Comment (thecogdoctor) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: If this breaks or could break something, it needs to be addressed before this PR goes forward.

Comment (PiZZAD0X) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: Do those functions exit on player/client? Could have set up two different systems for AI and Players. Should be all on XEH with a serverevent and isPlayer check

Comment (PiZZAD0X) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: Events were just blank if I remember correctly, so shouldnt have any effective change. They were for the future when server needed info on players dying/respawning

Comment (thecogdoctor) [reply uo_sys_framework/Core/cfgXEH/Killed/CAManBaseKilled.sqf]: > 
> 
> Events were just blank if I remember correctly, so shouldnt have any effective change. They were for the future when server needed info on players dying/respawning

So sacher's changes to these events is good to go ahead with?

**Pull request merged**